mrp_client.py

```diff
@@ -22,33 +26,189 @@ def getnetmask(self, machine_name, interface_name='eth1'):
         ipgetter = IPGetter(self.url, self.token, machine_name, interface_name)
         return ipgetter.get_netmask()
 
+    def provision(self, machine_name, preseed_name, initrd_desc, kernel_desc,
+                  kernel_opts, arch, subarch):
+        try:
+            provisioner = MachineProvisioner(
+                self.url, self.token, machine_name)
+            provisioner.set_machine_parameters(preseed_name, initrd_desc,
+                                               kernel_desc, kernel_opts, arch,
+                                               subarch)
+            provisioner.machine_provision()
+            return True
+        except ProvisionerError as err:
+            print(err)
+
+    def setparams(self, machine_name, preseed_name, initrd_desc, kernel_desc,
+                  kernel_opts, arch, subarch):
+        try:
+            provisioner = MachineProvisioner(
+                self.url, self.token, machine_name)
+            provisioner.set_machine_parameters(preseed_name, initrd_desc,
+                                               kernel_desc, kernel_opts, arch,
+                                               subarch)
+        except ProvisionerError as err:
+            print(err)
+
+    def im_upload(self, image_type, desc, arch, path, public, knowngood):
+        try:
+            im_uploader = ImageUploader(self.url, self.token)
+            print(im_uploader.upload_image(image_type, desc, arch, path, public,
+                                           knowngood))
+            return True
+        except ProvisionerError as err:
+            print(err)
+
+    def im_check(self, image_type, desc, arch):
+        try:
+            im_uploader = ImageUploader(self.url, self.token)
+            image = im_uploader.check_existence(image_type, desc, arch)
+            if not image:
+                print('Image does not exist on MrP')
```

Comment:
We need a better control, or it will be hard to embed this script in other scripts (including Ansible).
- Standard output, easily parsed
- Return non-zero for cases that may need more action (for example, "image not found")
- Maybe align with whatever MrP outputs in its API